JavaScript/JavaScriptSDK.Tests/CheckinTests/Context/Session.tests.ts

```diff
@@ -120,6 +128,88 @@ class SessionContextTests extends TestClass {
         });
 
         this.testCase({
+            name: "ai_session local storage has correct structure",
+            test: () => {
+                if (window.localStorage) {
+                    // setup
+                    var actualCookieName: string;
+                    var actualCookieValue: string;
+                    var newGuidStub = sinon.stub(Microsoft.ApplicationInsights.Util, "newGuid",() => "newGuid");
+                    var getCookieStub = sinon.stub(Microsoft.ApplicationInsights.Util, "getCookie",() => "");
+                    var setCookieStub = sinon.stub(Microsoft.ApplicationInsights.Util, "setCookie",(cookieName, cookieValue) => { });
+
+                    // act
+                    var sessionManager = new Microsoft.ApplicationInsights.Context._SessionManager(null,() => { });
+                    sessionManager.update();
+                    sessionManager.backup();
+
+                    // verify
+                    Assert.ok(localStorage["ai_session"], "ai_session storage is set");
+
+                    Assert.equal(3, localStorage["ai_session"].split('|').length, "Cookie value before expiration should include user id, acq date and renew date");
+                    Assert.equal("newGuid", localStorage["ai_session"].split('|')[0], "First part of cookie value should be new user id guid");
+
+                    // cleanup
+                    getCookieStub.restore();
+                    setCookieStub.restore();
+                    newGuidStub.restore();
+                } else {
+                    // this might happen on IE when using a file:// url
+                    Assert.ok(true, "browser does not support local storage in current environment");
+                }
+            }
+        });
+
+        this.testCase({
+            name: "SessionContext: session manager can recover old session id and isFirst state from lost cookies when localStorage is available",
+            test: () => {
+                // A more sophisticated approach of emulating document.cookie is needed to support multiple cookies
+                Microsoft.ApplicationInsights.Util["document"] = this.generateEmulatedCookieStore();
```

Comment:
For a person looking at this test for the first time, it feels [obscure](http://xunitpatterns.com/Obscure%20Test.html) because it is long, complex. It also appears to contain [conditional test logic](http://xunitpatterns.com/Conditional%20Test%20Logic.html) and multiple test conditions. I'm wondering if it could be split in two or more smaller tests.